test/powershell/engine/ParameterBinding/ParameterBinding.Tests.ps1

```diff
@@ -256,6 +256,11 @@ Describe "Parameter Binding Tests" -Tags "CI" {
         DynamicParamTest -PipelineVariable bar | ForEach-Object { $bar } | Should -Be "hi"
     }
 
+    It 'Dynamic parameter is found even if globbed path does not exist' {
+        $guid = New-Guid
+        { Copy-Item "~\$guid*" -Destination ~ -ToSession $null } | Should -Throw -ErrorId 'ParameterArgumentValidationError'
```

Comment:
It is not correct test for the PR's issue.
```powershell
> copy-item c:\noexist -Destination C:\temp\ -ToSession $null

Copy-Item: Cannot validate argument on parameter 'ToSession'. The argument is null or empty. Provide an argument that is not null or empty, and then try the command again.
```

Reply:
Perhaps I should add a comment to the test to clarify.  What this is validating is an error coming from the provider and not the parameter binder (which was the original issue)

Reply:
Without the fix I get `ParameterArgumentValidationError`.

Reply:
@iSazonov did you try this on Windows?  It seems to only repro on Windows now that I tried it on macOS

Reply:
Yes, on Windows.

Reply:
Did you make sure to include a wildcard?

```console
PS> copy-item c:\foo* -Destination ~ -tosession $null
Copy-Item: A parameter cannot be found that matches parameter name 'tosession'.
PS> copy-item c:\foo -Destination ~ -tosession $null
Copy-Item: Cannot validate argument on parameter 'ToSession'. The argument is null or empty. Provide an argument that is not null or empty, and then try the command again.
```

Reply:
Ah sorry, I copy-pasted the test with $guid null.